lib/jsdom/living/xhr/xhr-utils.js

```diff
@@ -269,26 +270,23 @@ function createClient(xhr) {
   }
 
   const options = {
-    uri,
     method: flag.method,
     headers: requestHeaders,
-    gzip: true,
-    maxRedirects: 21,
-    followAllRedirects: true,
-    encoding: null,
-    strictSSL: flag.strictSSL,
-    proxy: flag.proxy,
-    forever: true
+    agents,
+    rejectUnauthorized: flag.strictSSL,
+    followRedirects: true
   };
+
   if (flag.auth) {
-    options.auth = {
-      user: flag.auth.user || "",
-      pass: flag.auth.pass || "",
-      sendImmediately: false
-    };
+    const { user, pass } = flag.auth;
+    const username = typeof user === "string" ? user : "";
+    const password = typeof pass === "string" ? pass : "";
+    options.user = username;
+    options.pass = password;
```

Comment:
These lines could be much simpler as

```js
options.user = flag.auth.user || "";
options.pass = flag.auth.pass || "";
```

without four similarly-named intermediate variables.